STM/src/test/java/org/jboss/stm/TaxonomyNestedTest.java

```diff
@@ -0,0 +1,430 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, JBoss Inc., and individual contributors as indicated
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.stm;
+
+import com.arjuna.ats.arjuna.AtomicAction;
+import com.arjuna.ats.arjuna.coordinator.ActionStatus;
+import junit.framework.TestCase;
+import org.jboss.stm.annotations.Nested;
+import org.jboss.stm.annotations.ReadLock;
+import org.jboss.stm.annotations.Transactional;
+import org.jboss.stm.annotations.WriteLock;
+import org.jboss.stm.internal.RecoverableContainer;
+import org.junit.BeforeClass;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * There are no standards for STM and there are different models for how nested transactions behave.
+ * The following tests investigate the behaviour of the Narayana Software Transactional Memory
+ * implementation with respect to nested transactions.
+ *
+ * The tests are meant to supplement the STM discussion introduced
+ * <a href="https://jbossts.blogspot.com/2018/09/tips-on-how-to-evaluate-stm.html">in an online article</a>.
+ *
+ * The tests include:
+ *
+ * - checking that committing changes made by a closed nested transaction should be visible to the parent
```

Comment:
I would only mention that the javadoc is generated from html. It seems the markdown is used. The `-` will not be generated as list. The `<ul><li>` could be used for that being the list.

Reply:
Fixed